src/browser/api/window.js

```diff
@@ -1502,8 +1502,28 @@ Window.moveTo = function(identity, left, top) {
 };
 
 Window.navigate = function(identity, url) {
-    let browserWindow = getElectronBrowserWindow(identity, 'navigate');
-    browserWindow.webContents.loadURL(url);
+    return new Promise((resolve, reject) => {
+        let browserWindow = getElectronBrowserWindow(identity, 'navigate');
+
+        let handleLoadFinished = success => {
+            browserWindow.webContents.removeListener('did-fail-load', didFail);
+            browserWindow.webContents.removeListener('did-finish-load', didSucceed);
+            if (success) {
+                resolve();
+            } else {
+                reject();
+            }
+        };
+
+        let didFail = () => handleLoadFinished(false);
```

Comment:
Hey @tomer-openfin Looks like we are loosing some fidelity here, [Electron will return](https://electronjs.org/docs/api/web-contents#event-did-fail-load) the [Chromium net error](https://cs.chromium.org/chromium/src/net/base/net_error_list.h), take a look at how these errors are [handled in window creation](https://github.com/HadoukenIO/core/blob/01727fe61bb91994243ebe1fe45ad6f258b215da/src/browser/api/window.js#L776) and let's make sure we format our errors in a similar fashion.